annotation-processor/src/main/java/org/hibernate/validator/ap/checks/ConstraintCheckIssue.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.checks;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import java.util.Arrays;
+
+/**
+ * The result of the execution of a {@link ConstraintCheck}. Comprises
+ * information about the location at which the issue occurred and a message
+ * describing the occurred issue.
+ *
+ * @author Gunnar Morling
+ */
+public class ConstraintCheckIssue {
+
+	public enum IssueKind {
+		WARNING,
+		ERROR,
+	}
+
+	private final Element element;
+
+	private final AnnotationMirror annotationMirror;
+
+	private final IssueKind kind;
+
+	private final String messageKey;
+
+	private final Object[] messageParameters;
+
+	/**
+	 * Creates a new ConstraintCheckIssue.
+	 *
+	 * @param element The element at which the error occurred.
+	 * @param annotationMirror The annotation that causes the error.
+	 * @param messageKey A key for retrieving an error message template from the bundle
+	 * <p>
+	 * <code>org.hibernate.validator.ap.ValidationProcessorMessages.</code>
+	 * </p>
+	 *
+	 * @param messageParameters An array with values to put into the error message template
+	 * using {@link java.text.MessageFormat}. The number of elements must match
+	 * the number of place holders in the message template.
+	 */
+	public ConstraintCheckIssue(Element element, AnnotationMirror annotationMirror, IssueKind kind, String messageKey, Object... messageParameters) {
+
```

Comment:
wasn't me :) I just renamed the class 
I'll clean it up. But it got me thinking - maybe the style check can be added for such thing ? I remember that you've said that there's older code that uses such style. But maybe when the check is added all the code can be cleaned up ?
Or is that a bad thing to do ?